include/deal.II/lac/parpack_solver.h

```diff
@@ -516,6 +522,54 @@ void PArpackSolver<VectorType>::reinit(const dealii::IndexSet &locally_owned_dof
 }
 
 template <typename VectorType>
+void PArpackSolver<VectorType>::reinit(const IndexSet &locally_owned_dofs,
+                                       const std::vector<IndexSet> &block)
+{
+  // store local indices to write to vectors
+  locally_owned_dofs.fill_index_vector(local_indices);
+
+  // scalars
+  nloc = locally_owned_dofs.n_elements ();
+  ncv  = additional_data.number_of_arnoldi_vectors;
+
+  Assert ((int)local_indices.size() == nloc, ExcInternalError() );
+
+  // vectors
+  ldv = nloc;
+  v.resize (ldv*ncv, 0.0);
+
+  // TODO: add optional input for resid
+  resid.resize(nloc, 1.0);
+
+  // work arrays for ARPACK
+  workd.resize(3*nloc,0.0);
+
+  lworkl = additional_data.symmetric ?
+           ncv*ncv + 8*ncv
+           :
+           3*ncv*ncv+6*ncv;
+  workl.resize (lworkl, 0.);
+
+  ldz = nloc;
+  z.resize (ldz*ncv, 0.); // TODO we actually need only ldz*nev
+
+  // WORKEV  Double precision  work array of dimension 3*NCV.
+  lworkev = additional_data.symmetric ?
+            0 /*not used in symmetric case*/
+            :
+            3*ncv;
+  workev.resize (lworkev, 0.);
+
+  select.resize (ncv, 0);
+
+  // deal.II vectors:
+  src.reinit (block,mpi_communicator);
```

Comment:
everything but this line is the same for the two functions. I would suggest to refactor the code (maybe add a function in` private:` which is has the common part and then the two functions would call this private one.